code/game/objects/structures/table_frames.dm

```diff
@@ -83,26 +112,7 @@
 	framestack = /obj/item/stack/sheet/wood
 	framestackamount = 2
 	resistance_flags = FLAMMABLE
-
-/obj/structure/table_frame/wood/try_make_table(obj/item/stack/stack, mob/user)
-	if(!istype(stack, /obj/item/stack/tile/carpet) && !istype(stack, /obj/item/stack/sheet/wood))
-		return FALSE
-
-	if(stack.get_amount() < 1) //no need for safeties as we did an istype earlier
-		to_chat(user, "<span class='warning'>You need at least one sheet of [stack] to do this!</span>")
-		return TRUE
-
-	to_chat(user, "<span class='notice'>You start adding [stack] to [src]...</span>")
-
-	if(!(do_after(user, 50, target = src) && stack.use(1)))
-		return TRUE
-
-	if(istype(stack, /obj/item/stack/tile/carpet))
-		make_new_table(/obj/structure/table/wood/poker)
-		return TRUE
-
-	make_new_table(stack.table_type)
-	return TRUE
+	restrict_table_types = list(/obj/item/stack/tile/carpet = /obj/structure/table/wood/poker, /obj/item/stack/sheet/wood)
```

Comment:
you shouldn't mix associative lists with non-associative lists.
```suggestion
	restrict_table_types = list(/obj/item/stack/tile/carpet = /obj/structure/table/wood/poker, /obj/item/stack/sheet/wood = default/path/or/whatever)
```

Reply:
1. even though BYOND does not have hard typing, switching the type of values you use in a single var between children is not really good practice
2. You don't document the types of lists `restrict_table_types` is supposed to use, so if I look at the base var definition, I'm going to assume the correct input type is an assoc list or an empty list. This is confusing for future contributors and is undocumented behaviour.
3. The goal of a refactor like this is to make the code more efficient AND improve code clarity. So either document this more or simplify it.